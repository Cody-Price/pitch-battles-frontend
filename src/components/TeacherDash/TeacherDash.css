.teacher-dash {
  font-family: "Oswald", sans-serif;
  background: #f2f2f2;
  min-height: 100vh;
  color: #333;
}

.teacher-dash button,
.teacher-dash input {
  font-family: "Open Sans", sans-serif;
  background: #f2f2f2;
  color: #333;
}

.teacher-dash button {
  border-radius: 20px;
  border: none;

  background: rgba(41, 67, 199, 1);

  width: 70px;
  height: 30px;
  color: #f2f2f2;
  box-shadow: 2px 2px 2px rgba(0, 0, 0, 0.4);
  transition: 0.3s;
}

.teacher-dash button:hover {
  cursor: pointer;
  background: #da4a64;
}

.teacher-dash button:active {
  box-shadow: none;
}

.teacher-dash button:focus {
  outline: none;
}

.teacher-dash-header {
  box-shadow: 3px 3px 3px rgba(0, 0, 0, 0.2);

  position: relative;

  background: url(./images/dd-bg-long.png);
  background-size: cover;
  image-rendering: pixelated;

  color: #f2f2f2;
}

.teacher-dash-header h1 {
  font-family: "Open Sans", sans-serif;

  margin: 0;
  padding: 1rem 0 0.5rem 1rem;
  text-align: start;
  font-size: 1.6rem;
}

.teacher-dash-teacher-name {
  font-family: "Oswald", sans-serif;

  margin-top: 0;
  margin-left: 1rem;
  padding-bottom: 1rem;
  text-align: start;
  font-size: 1.3rem;
  font-weight: 700;
}

.classes-container {
  margin: 0 auto;
  display: flex;
  flex-wrap: wrap;
  justify-content: start;
  max-width: 820px;
}

.fas.fa-user,
.fas.fa-home {
  position: absolute;
  right: 1rem;
  top: 1rem;
  font-size: 1.5rem;
  transition: 0.3s;
}

.fas.fa-user:hover,
.fas.fa-home:hover {
  cursor: pointer;
  transform: scale(1.2);
}

.fas.fa-home {
  right: 4rem;
}

.create-new-class-form input {
  margin-top: 1rem;
}

.teacher-dash-filter {
  background: rgba(0, 0, 0, 0.5);
  width: 100%;
}
