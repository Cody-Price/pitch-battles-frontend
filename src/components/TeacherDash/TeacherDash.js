import React, { Component } from "react";
import "./TeacherDash.css";
import PropTypes from "prop-types";
import ClassCard from "../ClassCard/ClassCard";
import mockClassGroup from "../../utilities/mockClassGroup.js";
import TeacherClassView from "../TeacherClassView/TeacherClassView.js";

import { teacherAllClassesFetch } from "../../utilities/fetchCalls";

class TeacherDash extends Component {
  constructor() {
    super();
    this.state = {
      currentPage: "class cards",
      classes: [],
      error: false,
      selectedClass: null
    };
  }

  componentDidMount() {
    this.generateClassCards();
  }

  generateClassCards = async () => {
    try {
      const response = await teacherAllClassesFetch(this.props.webToken);

      console.log(response);
      this.setState({
        classes: response.data
      });
    } catch (error) {
      this.setState({
        error: true
      });
    }
  };

  navigate = currentPage => {
    this.setState({ currentPage });
  };

  navigateToClassPage = () => {
    this.navigate("selected class");
  };

  classSelect = id => {
    this.setState(
      {
        selectedClass: id
      },
      this.navigateToClassPage
    );
  };

  render() {
    const classes = this.state.classes.map(klass => {
      return <ClassCard data={klass} classSelect={this.classSelect} />;
    });
    return (
      <main className="teacher-dash">
        <header className="teacher-dash-header">
          <h1>Pitch Battles for Teachers</h1>

          <aside className="teacher-account-icon">
            <div className="teacher-dash-icon">
              <p className="teacher-dash-teacher-name">Kevin Simpson</p>
            </div>
          </aside>
          <div className="student-search" />
          <i
            className="fas fa-user"
            onClick={() => this.navigate("teacher account")}
          />
        </header>
        {this.state.currentPage === "class cards" && (
          <section className="classes-container">
            {classes}
            <article className="new-class-card">
              <h4 className="new-class-card-label">Create New Class</h4>
              <form
                className="create-new-class-form"
                onSubmit={e => this.handleSubmit(e)}
              >
                <input
                  name="newClass"
                  value={this.state.newClass}
                  placeholder="class name"
                  onChange={e => this.handleChange(e)}
                />
                <button className="new-class-button">create</button>
              </form>
            </article>
          </section>
        )}
        {this.state.currentPage === "selected class" && (
          <TeacherClassView
            id={this.state.selectedClass}
            webToken={this.props.webToken}
          />
        )}
        {this.state.currentPage === "teacher account" && <div />}
      </main>
    );
  }
}

export default TeacherDash;
