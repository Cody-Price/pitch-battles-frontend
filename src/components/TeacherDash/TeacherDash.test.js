import React from "react";
import ReactDOM from "react-dom";
import TeacherDash from "./TeacherDash";
import Enzyme, { shallow } from "enzyme";
const EnzymeAdapter = require("enzyme-adapter-react-16");

Enzyme.configure({ adapter: new EnzymeAdapter() });

describe("TeacherDash", () => {
  const mockUser = {
    attributes: {
      first_name: "Haley",
      last_name: "Jacobs"
    }
  };
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<TeacherDash user={mockUser} />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
