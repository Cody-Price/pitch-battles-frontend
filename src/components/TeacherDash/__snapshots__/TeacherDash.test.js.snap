// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeacherDash should match the snapshot 1`] = `
<main
  className="teacher-dash"
>
  <header
    className="teacher-dash-header"
  >
    <div
      className="teacher-dash-filter"
    >
      <h1>
        Pitch Battles for Teachers
      </h1>
      <aside
        className="teacher-account-icon"
      >
        <div
          className="teacher-dash-icon"
        >
          <p
            className="teacher-dash-teacher-name"
          >
            Haley
             
            Jacobs
          </p>
        </div>
      </aside>
      <div
        className="student-search"
      />
      <i
        className="fas fa-home"
        onClick={[Function]}
      />
      <i
        className="fas fa-user"
        onClick={[Function]}
      />
    </div>
  </header>
  <section
    className="classes-container"
  >
    <article
      className="new-class-card"
    >
      <header>
        <div
          className="class-card-header-tint"
        >
          <h4
            className="new-class-card-label"
          >
            Create New Class
          </h4>
        </div>
      </header>
      <form
        className="create-new-class-form"
        onSubmit={[Function]}
      >
        <input
          name="newClassName"
          onChange={[Function]}
          placeholder="class name"
          value=""
        />
        <button
          className="new-class-button"
        >
          create
        </button>
      </form>
    </article>
  </section>
</main>
`;
